src/pd/client.rs

```diff
@@ -72,17 +76,81 @@ fn parse_urls(addr: &str) -> Result<Vec<String>> {
     Ok(hosts)
 }
 
-fn rpc_connect(addr: &str) -> Result<TcpStream> {
-    let hosts = try!(parse_urls(addr));
+fn parse_pd_members(json: &Json) -> Result<Vec<String>> {
+    let mut hosts = Vec::new();
+
+    if let Some(&Json::Array(ref members)) = json.find("members") {
+        for m in members {
+            if let Some(&Json::Array(ref client_urls)) = m.find("client-urls") {
+                for url in client_urls {
+                    if let &Json::String(ref u) = url {
+                        hosts.extend(try!(parse_urls(u)))
+                    }
+                }
+            }
+        }
+    }
+
+    if hosts.len() > 0 {
+        Ok(hosts)
+    } else {
+        Err(box_err!("invalid pd members object"))
+    }
+}
+
+fn get_pd_members(endpoints: &str) -> Result<Vec<String>> {
```

Comment:
This is a rare event, e,g, if we pass 3 PD endpoints, we can't remove this 3 PD and add another 3 PD, this is crazy in production. 

Btw, we can get all members in RPC, so maybe we can first choose an endpoint to connect, and then get all members. But I think we can support this later, now using argument passing is ok. 

I still think whether we should get leader or not first, or use proxy directly? how do you think?